### PR TITLE
Introduce the `count_distinct` aggregation function

### PR DESCRIPTION
**Pull request by dominiklohmann**
`count` and `distinct`, fused together. This new aggregation function returns the number of distinct, non-null values.

In addition, this removes the special-handling of list values in distinct: The `distinct` function silently performed a different operation on lists, returning the distinct non-null elements in the list rather than operating on the list itself. This feature will return in the future as unnesting on the extractor level via `distinct(field[])`, but for now it has to go to make the distinct aggregation function work consistently.

### CODE REVIEW

File: libvast/test/summarize.cpp [hunk 1]
```diff
@@ -198,14 +190,11 @@ TEST(summarize test) {
               vast::time{std::chrono::seconds(1258329609)});
   const auto expected_ports = list{uint64_t{443}};
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/test/summarize.cpp:191]: I swapped the test around a bit to have a set of size three in 2cfade887. Does this work for you?

[the remark this answers is not in the document]
Comment (mavam) [reply libvast/test/summarize.cpp]: Almost, I still want the dedup'ing aspect somewhere in the test.

[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/test/summarize.cpp]: That is deduping, there are 10 values of which three are distinct and non-null in that column.

[the remark this answers is not in the document]
Comment (mavam) [reply libvast/test/summarize.cpp]: Ah, dang mobile review! 🙈

**Pull request merged**